sdk/attestation/attestation/src/utils/utf8.browser.ts

```diff
@@ -1,55 +1,38 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-/// <reference lib="dom" />
-
-import { TextDecoder, TextEncoder } from "./textEncoding";
-
-let encoder: TextEncoder | undefined;
-let decoder: TextDecoder | undefined;
-
-/**
- * Returns a cached TextEncoder.
- * @internal
- */
-function getTextEncoder(): TextEncoder {
-  if (encoder) {
-    return encoder;
-  }
-
-  if (typeof TextEncoder === "undefined") {
-    throw new Error(`Your browser environment is missing "TextEncoder".`);
-  }
-
-  encoder = new TextEncoder();
-  return encoder;
-}
+declare let TextDecoder: undefined | (new() => { decode(buffer: ArrayBuffer | ArrayBufferView): string});
+declare let TextEncoder:  undefined | (new() => { encode(str : string): ArrayBuffer});
+
+// TextDecoder and TextEncoder are in the global namespace for Node version 11 and
+// higher, but before that, they were in the "util" namespace. If we're running
+// under node ("Buffer" is defined), then check to see if the global namespace version
+// of the decoders are present, if not, import them from the util namespace.
+const decoder = typeof Buffer === "undefined" ?
+  new (typeof TextDecoder !== "undefined" ? TextDecoder : require("util").TextDecoder)("ascii") :
+  undefined;
```

Comment:
@witemple-msft @xirzec what do you think of this pattern? seems like the pros is we do not need an extra file and a mapping in package.json and the cons is the need to do dynamic loading which is anti-pattern in ts.

Reply:
FWIW, this was suggested by @DanielRosenwasser.

Reply:
This isn't quite a dynamic require, and is probably fine with our bundling since it is a node builtin module. I'm good with this, except I would suggest `new (TextDecoder ?? require("util").TextDecoder)("ascii")` instead of the ternary.